server/datastore/mysql/hosts.go

```diff
@@ -4361,6 +4364,45 @@ func (ds *Datastore) UpdateHost(ctx context.Context, host *fleet.Host) error {
 	return nil
 }
 
+func (ds *Datastore) OSVersion(ctx context.Context, osVersionID uint, teamID *uint) (*fleet.OSVersion, *time.Time, error) {
+	jsonValue, updatedAt, err := ds.executeOSVersionQuery(ctx, teamID)
```

Comment:
Can't you use SQL to filter by osVersionID and grab the one entry (or entries) you need?
https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html

Reply:
I played around with this for a bit and haven't gotten to a viable solution yet.  Seems like JSON_SEARCH may have difficulty with non-string values, but the query looks something like this, which may be confusing to maintain:
```
SELECT JSON_UNQUOTE(JSON_EXTRACT(json_value, CONCAT('$[', JSON_SEARCH(json_value, 'one', ?, NULL, '$[*].os_version_id'), ']'))) AS json_object
              FROM aggregated_stats
              WHERE JSON_SEARCH(json_value, 'one', ?, NULL, '$[*].os_version_id') IS NOT NULL
```